cube/bmp_8.cpp

```diff
@@ -77,14 +73,14 @@ unsigned char **Bmp8::CreateBmpBuffer(unsigned char init_val) {
 
   buff = (unsigned char **) new unsigned char *[hgt_ * sizeof(*buff)];
   if (!buff) {
```

Comment:
* Do any special build options matter for the handling of C++ exceptions by such a software component?
* Can any smart pointers help to improve the source code further?